tests/php/ORM/DataObjectTest.php

```diff
@@ -2671,4 +2672,36 @@ public function testDBObjectEnum()
         $vals = ['25.25', '50.00', '75.00', '100.50'];
         $this->assertSame(array_combine($vals ?? [], $vals ?? []), $obj->dbObject('MyEnumWithDots')->enumValues());
     }
+
+    public function testGetDatabaseBackedField()
+    {
+        $dataObjectClass = new DataObject();
+        $method = new ReflectionMethod($dataObjectClass, 'getDatabaseBackedField');
+        $method->setAccessible(true);
+        $class = new Team([]);
+
+        $databaseBackedField = $method->invokeArgs($class, ['Captain.FavouriteTeam.Title']);
+        $this->assertSame('Captain.FavouriteTeam.Title', $databaseBackedField);
+
+        $databaseBackedField = $method->invokeArgs($class, ['Captain.ShirtNumber']);
+        $this->assertSame('Captain.ShirtNumber', $databaseBackedField);
+
+        $databaseBackedField = $method->invokeArgs($class, ['Captain.FavouriteTeam.ReturnsNull']);
+        $this->assertSame('Captain.FavouriteTeam', $databaseBackedField);
```

Comment:
This one should return null - `FavouriteTeam` isn't a database backed field, it's a whole DataObject record.

The following is meant to handle this scenario:
```php
if ($component->hasMethod($nextPart)) {
    // If the next part is a method, we don't have a database-backed field.
    return null;
}
```

Reply:
Updated